zerver/lib/message.py

```diff
@@ -1220,21 +1220,23 @@ def check_user_group_mention_allowed(sender: UserProfile, user_group_ids: List[i
 
     for group in user_groups:
         can_mention_group = group.can_mention_group
-        can_mention_group_name = can_mention_group.named_user_group.name
+        if (
+            hasattr(can_mention_group, "named_user_group")
+            and can_mention_group.named_user_group.name == SystemGroups.EVERYONE
+        ):
+            continue
         if sender_is_system_bot:
-            if can_mention_group_name == SystemGroups.EVERYONE:
-                continue
             raise JsonableError(
-                _(
-                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of '{can_mention_group_name}' to mention this group."
-                ).format(user_group_name=group.name, can_mention_group_name=can_mention_group_name)
+                _("You are not allowed to mention user group '{user_group_name}'.").format(
+                    user_group_name=group.name
+                )
             )
 
         if not is_user_in_group(can_mention_group, sender, direct_member_only=False):
             raise JsonableError(
-                _(
-                    "You are not allowed to mention user group '{user_group_name}'. You must be a member of '{can_mention_group_name}' to mention this group."
-                ).format(user_group_name=group.name, can_mention_group_name=can_mention_group_name)
+                _("You are not allowed to mention user group '{user_group_name}'.").format(
```

Comment:
Updated the error message to not include "You must be a member of ..." part as discussed in #29124.

Reply:
Yeah, this is great, just simplifies the strings.